### PR TITLE
Update banner and page to follow GDS cookie guidance

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

```
As a... candidate using Apply or provider using Manage
I need to... decide which cookies are are place on my computer
So that... I can opt in of out of having my activity tracked
```

We currently have different implementations for our cookie banners across Find and Apply. These also are somewhat nefarious as they make it more difficult to reject cookies (you need to visit the cookies page to change this setting).

## Changes proposed in this pull request

* The cookie banner uses the markup and styles as outlined in the design system
* The cookie banner component follows [the guidance from GDS](https://design-system.service.gov.uk/components/cookie-banner/)
* The cookie page follows [the guidance from GDS for this pattern](https://design-system.service.gov.uk/patterns/cookies-page/)

cookies

## Guidance to review

- Go to `/` or `/provider` to view changes to the cookie banner.
- Go to `/candidate/cookies` and `/provider/cookies` to see changes to cookie guidance
- Note that unlike Find and GiT, this implementation is sans-JS. The intention is to move to a JS implementation in future (@paulrobertlloyd  to confirm)

## Link to Trello card

https://trello.com/c/3wlz7cyM/3061-%F0%9F%8D%AA-update-banner-and-page-to-follow-gds-cookie-guidance-apply

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by despo — APPROVED** [review]

File: app/views/content/cookies.html.erb [hunk 1]
```diff
@@ -1,140 +1,138 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
-    <h1 class="govuk-heading-xl">Cookies</h1>
-
-    <h2 class="govuk-heading-l" id="cookies-that-measure-website-use">Cookies that measure website use</h2>
-
-    <p class="govuk-body">
-    We use Google Analytics software to collect information about how you use ‘<%= service_name %>’. We do this to help make sure the site is meeting the needs of its users and to help us make improvements.
+    <h1 class="govuk-heading-l">Cookies</h1>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:3]: We’re currently using `xl` for headings on Apply, so we can revert this change.

```suggestion
    <h1 class="govuk-heading-xl">Cookies</h1>
```

File: app/views/content/cookies.html.erb [hunk 2]
```diff
@@ -1,140 +1,138 @@
...
+    <p class='govuk-body'>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:4]: Use double quotes for HTML attributes.

```suggestion
    <p class="govuk-body">
```

File: app/views/content/cookies.html.erb [hunk 3]
```diff
@@ -1,140 +1,138 @@
...
-    <p id="preferences" class="govuk-body">
-    <%= form_with model: @cookie_preferences, url: @cookie_settings_path, method: :post do |f| %>
-    <%= f.govuk_collection_radio_buttons :consent,
-      [['yes', 'Yes, opt-in to Google Analytics cookies'], ['no', 'No, do not track my website usage']],
-      :first,
-      :last,
-      legend: { text: 'Do you accept Google Analytics cookies?' } %>
-    <%= f.govuk_submit 'Save preferences' %>
-    <% end %>
+    <p class='govuk-body'>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:7]:
```suggestion
    <p class="govuk-body">
```

File: app/views/content/cookies.html.erb [hunk 4]
```diff
@@ -1,140 +1,138 @@
...
+      Google is not allowed to use or share our analytics data with anyone.
     </p>
+    <p class="govuk-body">
+      Google Analytics stores anonymised information about:
+    </p>
+
+    <ul class="govuk-list govuk-list--bullet">
+      <li>how you got to ‘<%= service_name %>’</li>
+      <li>the pages you visit on this site and how long you spend on them</li>
+      <li>what you click on while you're visiting the site</li>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:72]: 👀

```suggestion
      <li>what you click on while you’re visiting the site</li>
```

File: app/views/shared/_cookies_banner.html.erb [hunk 5]
```diff
@@ -1,15 +1,35 @@
...
-    <% end %>
-  </div>
-</div>
+<%= govuk_cookie_banner(
+  title: "Cookies on ‘#{service_name}’",
+  html_attributes: {
+    aria: {
+      label: "Cookies on ‘#{service_name}’",
+    },
+  },
```
Comment (paulrobertlloyd) [inline app/views/shared/_cookies_banner.html.erb:7]:
```suggestion
  title: "Cookies on #{service_name}",
  html_attributes: {
    aria: {
      label: "Cookies on #{service_name}",
    },
  },
```

File: app/views/shared/_cookies_confirmation_banner.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<%= govuk_cookie_banner(
+      html_attributes: {
+        aria: {
+          label: "Cookies on ‘#{service_name}’",
```
Comment (paulrobertlloyd) [inline app/views/shared/_cookies_confirmation_banner.html.erb:4]:
```suggestion
          label: "Cookies on #{service_name}",
```

File: spec/system/candidate_interface/candidate_content_spec.rb [hunk 7]
```diff
@@ -38,7 +38,7 @@ def then_i_can_see_the_accessibility_statement
   end
 
   def and_i_can_see_the_cookie_banner
-    expect(page).to have_content('We use cookies to collect information about how you use ‘Apply for teacher training’')
+    expect(page).to have_content('Cookies on ‘Apply for teacher training’')
```
Comment (paulrobertlloyd) [inline spec/system/candidate_interface/candidate_content_spec.rb:41]:
```suggestion
    expect(page).to have_content('Cookies on Apply for teacher training')
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
First pass review. Small markup tweaks and removing smart quotes form service name in the banner.

File: app/views/content/cookies.html.erb [hunk 8]
```diff
@@ -1,140 +1,138 @@
...
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Purpose
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Expires
+        </th>
+      </tr>
       </thead>
       <tbody class="govuk-table__body govuk-!-font-size-14">
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:31]: Let’s not use small text in these tables. 

```suggestion
      <tbody class="govuk-table__body">
```

File: app/views/content/cookies.html.erb [hunk 9]
```diff
@@ -1,140 +1,138 @@
...
-          <td class="govuk-table__cell">
-            Saves your cookie consent settings
-          </td>
-          <td class="govuk-table__cell">
-            6 months
-          </td>
-        </tr>
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell">
+          _apply_for_postgraduate_teacher_training_session
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:34]: Given the length of this string, maybe we can force a line break in the middle:

```suggestion
          _apply_for_postgraduate<wbr>_teacher_training_session
```

File: app/views/content/cookies.html.erb [hunk 10]
```diff
@@ -1,140 +1,138 @@
...
+      <tr class="govuk-table__row">
+        <th scope="col" class="govuk-table__header">
+          Name
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Purpose
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Expires
+        </th>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:28]: Same layout as on Find

```suggestion
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Name
        </th>
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Purpose
        </th>
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Expires
        </th>
```

File: app/views/content/cookies.html.erb [hunk 11]
```diff
@@ -1,140 +1,138 @@
...
+      </tr>
       </thead>
       <tbody class="govuk-table__body govuk-!-font-size-14">
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:89]: As with the other table on this page

```suggestion
      <tbody class="govuk-table__body">
```

File: app/views/content/cookies.html.erb [hunk 12]
```diff
@@ -1,140 +1,138 @@
...
+      <tr class="govuk-table__row">
+        <th scope="col" class="govuk-table__header">
+          Name
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Purpose
+        </th>
+        <th scope="col" class="govuk-table__header">
+          Expires
+        </th>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:86]: Same as other table on this page

```suggestion
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Name
        </th>
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Purpose
        </th>
        <th scope="col" class="govuk-table__header govuk-!-width-one-third">
          Expires
        </th>
```

File: app/views/content/cookies.html.erb [hunk 13]
```diff
@@ -1,140 +1,138 @@
...
-          <td class="govuk-table__cell">
-            10 minutes
-          </td>
-        </tr>
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell">
+          _ga
+        </td>
+        <td class="govuk-table__cell">
+          Used to distinguish anonymous users
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:95]:
```suggestion
          Distinguishes anonymous users
```

File: app/views/content/cookies.html.erb [hunk 14]
```diff
@@ -1,140 +1,138 @@
...
+        <td class="govuk-table__cell">
+          2 years
+        </td>
+      </tr>
+      <tr>
+        <td class="govuk-table__cell">
+          _gid
+        </td>
+        <td class="govuk-table__cell">
+          Used to distinguish anonymous users
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:106]:
```suggestion
          Distinguishes anonymous users
```

File: app/views/content/cookies.html.erb [hunk 15]
```diff
@@ -1,140 +1,138 @@
...
+        <td class="govuk-table__cell">
+          24 hours
+        </td>
+      </tr>
+      <tr>
+        <td class="govuk-table__cell">
+          _gat_gtag_<%= @cookie_ga_code %>
+        </td>
+        <td class="govuk-table__cell">
+          Throttle requests
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:117]:
```suggestion
          Limits requests
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Content changes to cookies page to align with that used on Find.

File: app/views/content/cookies.html.erb (hunk 2 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:4]: 🤔 Interesting that the erb linter does not pick up on this

File: app/views/content/cookies.html.erb (hunk 1 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:3]: 👍

File: app/views/content/cookies.html.erb (hunk 3 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:7]: 👍

File: app/views/content/cookies.html.erb (hunk 4 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:72]: 👍

File: app/views/shared/_cookies_banner.html.erb (hunk 5 above)
Comment (TobyRet) [reply app/views/shared/_cookies_banner.html.erb:7]: 👍

File: app/views/shared/_cookies_confirmation_banner.html.erb (hunk 6 above)
Comment (TobyRet) [reply app/views/shared/_cookies_confirmation_banner.html.erb:4]: 👍

File: app/views/content/cookies.html.erb (hunk 8 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:31]: 👍

File: app/views/content/cookies.html.erb (hunk 2 above)
Comment (paulrobertlloyd) [reply app/views/content/cookies.html.erb:4]: Yeah, that was my hope, somewhat dashed by reality.

File: app/views/content/cookies.html.erb (hunk 9 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:34]: 👍

File: app/views/content/cookies.html.erb (hunk 10 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:28]: 👍

File: app/views/content/cookies.html.erb (hunk 12 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:86]: 👍

File: app/views/content/cookies.html.erb (hunk 15 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:117]: 👍

File: app/views/content/cookies.html.erb (hunk 13 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:95]: 👍

File: app/views/content/cookies.html.erb (hunk 14 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:106]: 👍

File: app/views/content/cookies.html.erb (hunk 11 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:89]: 👍

File: app/frontend/styles/_button.scss [hunk 16]
```diff
@@ -1,4 +1,5 @@
-// TODO: Remove once implemented cookie banner from design system
-.app-button--cookies {
-  min-width: 12em;
+.app-button-group--align-center {
+  .govuk-button-group {
+    align-items: center;
+  }
```
Comment (paulrobertlloyd) [inline app/frontend/styles/_button.scss:4]:
```suggestion
// TODO: Remove once cookie banner implemented using JS (i.e. without `button_to` form buttons)
.govuk-cookie-banner .govuk-button-group {
  align-items: center;
```

Comment (paulrobertlloyd) [reply app/frontend/styles/_button.scss:4]: …and we should rename this file `_cookie-banner.scss`

File: app/views/content/cookies.html.erb [hunk 17]
```diff
@@ -1,140 +1,138 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
     <h1 class="govuk-heading-xl">Cookies</h1>
-
-    <h2 class="govuk-heading-l" id="cookies-that-measure-website-use">Cookies that measure website use</h2>
-
-    <p class="govuk-body">
-    We use Google Analytics software to collect information about how you use ‘<%= service_name %>’. We do this to help make sure the site is meeting the needs of its users and to help us make improvements.
+    <p class='govuk-body'>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:4]: Not sure if reviewing older commit?

```suggestion
    <p class="govuk-body">
```

File: app/views/content/cookies.html.erb [hunk 18]
```diff
@@ -1,140 +1,138 @@
...
+    <%= form_with model: @cookie_preferences, url: @cookie_settings_path, method: :post do |f| %>
+      <%= f.govuk_collection_radio_buttons :consent,
+        [%w[yes Yes], %w[no No]],
+        :first,
+        :last,
+        inline: true,
+        legend: { text: 'Do you want to accept Google Analytics cookies?', size: 's' } %>
+      <%= f.govuk_submit 'Save cookie settings' %>
+      <% end %>
+    </p>
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:136]: Do you need the `preferences` id? If so, can add it to the form.

```suggestion
    <%= form_with model: @cookie_preferences, url: @cookie_settings_path, method: :post do |f| %>
      <%= f.govuk_collection_radio_buttons :consent,
        [%w[yes Yes], %w[no No]],
        :first,
        :last,
        inline: true,
        legend: { text: 'Do you want to accept Google Analytics cookies?', size: 's' } %>
      <%= f.govuk_submit 'Save cookie settings' %>
    <% end %>
```

File: app/views/shared/_cookies_banner.html.erb [hunk 19]
```diff
@@ -1,15 +1,36 @@
...
+  html_attributes: {
+    aria: {
+      label: "Cookies on #{service_name}",
+    },
+  },
+) do |cookie_banner| %>
+<%= cookie_banner.with(:body) do %>
+  <p class="govuk-body">We use some essential cookies to make this service work.</p>
+  <p class="govuk-body">We’d also like to use analytics cookies so we can understand how you use the service and make
+  improvements.</p>
```
Comment (paulrobertlloyd) [inline app/views/shared/_cookies_banner.html.erb:13]:
```suggestion
  <p class="govuk-body">We’d also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
```

File: app/views/shared/_cookies_confirmation_banner.html.erb [hunk 20]
```diff
@@ -0,0 +1,14 @@
+<%= govuk_cookie_banner(
+      html_attributes: {
+        aria: {
+          label: "Cookies on #{service_name}",
+        },
+      },
+    ) do |cookie_banner| %>
```
Comment (paulrobertlloyd) [inline app/views/shared/_cookies_confirmation_banner.html.erb:7]: Probably use this indentation:

```suggestion
<%= govuk_cookie_banner(
  html_attributes: {
    aria: {
      label: "Cookies on #{service_name}",
    },
  },
) do |cookie_banner| %>
```

File: app/views/content/cookies.html.erb [hunk 21]
```diff
@@ -1,140 +1,138 @@
...
+        <td class="govuk-table__cell">
+          24 hours
+        </td>
+      </tr>
+      <tr>
+        <td class="govuk-table__cell">
+          _gat_gtag_<%= @cookie_ga_code %>
+        </td>
+        <td class="govuk-table__cell">
+          Limit requests
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:117]:
```suggestion
          Limits requests
```

File: app/views/content/cookies.html.erb [hunk 22]
```diff
@@ -1,140 +1,138 @@
...
+        <td class="govuk-table__cell">
+          2 years
+        </td>
+      </tr>
+      <tr>
+        <td class="govuk-table__cell">
+          _gid
+        </td>
+        <td class="govuk-table__cell">
+          Distinguish anonymous users
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:106]:
```suggestion
          Distinguishes anonymous users
```

File: app/views/content/cookies.html.erb [hunk 23]
```diff
@@ -1,140 +1,138 @@
...
-            10 minutes
-          </td>
-        </tr>
+      <tbody class="govuk-table__body">
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell">
+          _ga
+        </td>
+        <td class="govuk-table__cell">
+          Distinguish anonymous users
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:95]:
```suggestion
          Distinguishes anonymous users
```

File: app/views/content/cookies.html.erb [hunk 24]
```diff
@@ -1,140 +1,138 @@
...
-    <% end %>
-    </p>
-
-    <h2 class="govuk-heading-l" id="how-we-use-cookies">How we use cookies</h2>
 
     <h3 class="govuk-heading-m" id="essential-cookies">Essential cookies</h3>
 
     <p class="govuk-body">
-    Essential ‘<%= service_name %>’ cookies include:
+      Essential ‘<%= service_name %>’ cookies keep your information secure while you use this service. We do not need to ask permission to use them.
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:14]:
```suggestion
      Essential cookies keep your information secure while you use ‘<%= service_name %>’. We do not need to ask permission to use them.
```

File: app/views/content/cookies.html.erb [hunk 25]
```diff
@@ -1,140 +1,138 @@
...
-
-    <p class="govuk-body">
-    We use Google Analytics software to collect information about how you use ‘<%= service_name %>’. We do this to help make sure the site is meeting the needs of its users and to help us make improvements.
+    <p class='govuk-body'>
+      Cookies are small files saved on your phone, tablet or computer when you visit a website.
     </p>
-
     <p class="govuk-body">
-      Google Analytics stores information about:
+      We use cookies to make this site work and collect information about how you use our service.
```
Comment (paulrobertlloyd) [inline app/views/content/cookies.html.erb:8]:
```suggestion
      We use cookies to make ‘<%= service_name %>’ work and collect information about how you use our service.
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Last few set of content and markup issues, then good to go, I think.

File: app/frontend/styles/_button.scss (hunk 16 above)
Comment (TobyRet) [reply app/frontend/styles/_button.scss:4]: 👍

File: app/views/content/cookies.html.erb (hunk 17 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:4]: 🤔 Thought I fixed that sorry

File: app/views/content/cookies.html.erb (hunk 18 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:136]: 👍

File: app/views/shared/_cookies_confirmation_banner.html.erb (hunk 20 above)
Comment (TobyRet) [reply app/views/shared/_cookies_confirmation_banner.html.erb:7]: 👍

File: app/views/content/cookies.html.erb (hunk 22 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:106]: 👍

File: app/views/content/cookies.html.erb (hunk 23 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:95]: 👍

File: app/views/content/cookies.html.erb (hunk 25 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb:8]: 👍

File: app/views/shared/_cookies_banner.html.erb (hunk 19 above)
Comment (TobyRet) [reply app/views/shared/_cookies_banner.html.erb:13]: 👍

File: app/views/content/cookies.html.erb (hunk 21 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb]: 👍

File: app/views/content/cookies.html.erb (hunk 24 above)
Comment (TobyRet) [reply app/views/content/cookies.html.erb]: 👍

**Review by paulrobertlloyd — APPROVED** [review]
